src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1319,7 +1319,7 @@ private BoundExpression SynthesizeMethodGroupReceiver(CSharpSyntaxNode syntax, A
             var declaringType = members[0].ContainingType;
 
             HashSet<DiagnosticInfo> unused = null;
-            if (currentType.IsEqualToOrDerivedFrom(declaringType, TypeCompareKind.ConsiderEverything, useSiteDiagnostics: ref unused))
+            if (currentType.IsInterface || currentType.IsEqualToOrDerivedFrom(declaringType, TypeCompareKind.ConsiderEverything, useSiteDiagnostics: ref unused))
```

Comment:
I don't think this is correct for an interface nested within a class that does not implement the interface. If we find a method invocation of an instance method of the enclosing class, this would appear to return an incompatible "this". The design of this method would appear to require that `null` be returned rather than an incompatible receiver. Here is the scenario:

``` c#
class A
{
    public void M() {}
    interface I
    {
        void N() { M(); } // what `this` here?
    }
}
```

I suspect the correct fix is to use something other than (or modify) `IsEqualToOrDerivedFrom`, which considers only non-interface inheritance.

The alternative is to remove the test for reference-relatedness from this method entirely and rely on a mechanism for that to be checked elsewhere, if you believe that such a larger refactoring has other benefits.

Please add tests for nested types, and specifically implicit receivers for invocations of names found on or inherited into enclosing types.